http-netty/src/main/java/io/micronaut/http/netty/NettyHttpRequestBuilder.java

```diff
@@ -74,12 +75,25 @@ default HttpRequest toHttpRequest() {
      * request has been changed, this will return an empty value.
      *
      * @return The request including the body
+     * @deprecated Go through {@link #toHttpRequestWithoutBody()} and {@link #byteBodyDirect()} instead
      */
+    @Deprecated
     @NonNull
     default Optional<HttpRequest> toHttpRequestDirect() {
         return Optional.empty();
     }
 
+    /**
+     * Directly convert this request body to a {@link ByteBody}, if possible. If the body of this
+     * request has been changed, this will return an empty value.
+     *
+     * @return The body
+     */
+    @NonNull
+    default Optional<ByteBody> byteBodyDirect() {
```

Comment:
For internal API I would just use nullable type

Reply:
Fair but I don't want to deal with merge conflicts now :D I will do it in a future PR

Reply:
included in #11182